### PR TITLE
Disjoint set

### PR DESCRIPTION
**Pull request by ahmed-irfan**

### CODE REVIEW

File: src/utils.cpp [hunk 1]
```diff
@@ -126,4 +126,94 @@ void get_free_symbolic_consts(const smt::Term &term, smt::TermVec &out)
...
+    {
+      Term leaderb = leader_.at(b);
+
+      if (leadera != leaderb)
+      {
+        UnorderedTermSet & groupb = group_.at(leaderb);
+
+        if (comp(leadera, leaderb))
+        {
+          // Choose according to the given Ranking
```
Comment (makaimann) [inline src/utils.cpp:153]:
```suggestion
          // Choose according to the given ranking
```
I don't feel that strongly about this btw, just wasn't sure that it needed to be capitalized.

**Review by makaimann — APPROVED** [review]
Looks great, thanks for adding this! I left a couple minor comments.

File: include/utils.h [hunk 2]
```diff
@@ -79,4 +79,29 @@ void disjunctive_partition(const smt::Term & term,
...
+ */
+class DisjointSet
+{
+ public:
+  DisjointSet(bool (*c)(const smt::Term & a, const smt::Term & b));
+  ~DisjointSet();
+
+  DisjointSet() = delete;
+
+  void add(const smt::Term & a, const smt::Term & b);
```
[the remark this answers is not in the document]
Comment (ahmed-irfan) [reply include/utils.h:97]: done in 9ff961e

**Comment by makaimann** [comment]
Thanks for the updates!

**Pull request merged**